schulcloud/AppDelegate.swift

```diff
@@ -47,20 +47,13 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
     
     /// Check for existing login credentials and return appropriate view controller
     func selectInitialViewController(application: UIApplication) -> UIViewController {
-        let defaults = UserDefaults.standard
-        
-        if let accountId = defaults.string(forKey: "accountId"),
-            let userId = defaults.string(forKey: "userId") {
-            var account = SchulCloudAccount(userId: userId, accountId: accountId, accessToken: nil)
-            account.loadAccessTokenFromKeychain()
-            if account.accessToken != nil {
-                return prepareInitialViewController(with: account)
-            } else {
-                log.error("Could not load account from Keychain!")
-            }
+        if let validAccount = LoginHelper.loadAccount() {
+            return prepareInitialViewController(with: validAccount)
+        } else {
+            log.info("Could not find existing login credentials, proceeding to login")
+            recreatePersistentContainer()
```

Comment:
On the first app launch, we recreate a brand new persistentContainer. Since the app launch seems sometimes slow to me, we should avoid such behavior.
Proposed solution: extract the account validation from `loadAccount`

Reply:
I think it would be more complicated to do this asynchronously or at a later point. How about we set a flag in the `UserDefaults` instead?

Reply:
There is not asynchronous code required. It would be something like this:
```swift
guard let account = loadAccount() else {
  // show login
}

guard validate(account) else {
  // reset db
  // show login
}

// skip login
```